File: Lesson14/app/resultapifragmentapp/src/main/java/ru/mirea/vasilev/resultapifragmentapp/DataFragment.java
package ru.mirea.vasilev.resultapifragmentapp;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

public class DataFragment extends Fragment {
    private EditText editTextInfo;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_data, container, false);

        editTextInfo = view.findViewById(R.id.editTextInfo);
        Button buttonOpenBottomSheet = view.findViewById(R.id.buttonOpenBottomSheet);

        buttonOpenBottomSheet.setOnClickListener(v -> {
            // Получение текста из EditText
            String info = editTextInfo.getText().toString();

            // Передача данных в BottomSheetFragment
            Bundle bundle = new Bundle();
            bundle.putString("info", info);

            BottomSheetFragment bottomSheetFragment = new BottomSheetFragment();
            bottomSheetFragment.setArguments(bundle);

            bottomSheetFragment.show(getParentFragmentManager(), "BottomSheetFragment");
        });

        return view;
    }
}

File: MireaProject/app/src/main/java/ru/mirea/vasilevmn/mireaproject/ui/data/DataFragment.kt
package ru.mirea.vasilevmn.mireaproject.ui.data

import androidx.lifecycle.ViewModelProvider
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import ru.mirea.vasilevmn.mireaproject.R
import ru.mirea.vasilevmn.mireaproject.databinding.FragmentDataBinding

class DataFragment : Fragment() {

    private var _binding: FragmentDataBinding? = null

    // This property is only valid between onCreateView and
    // onDestroyView.
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        val dataViewModel =
            ViewModelProvider(this)[DataViewModel::class.java]

        _binding = FragmentDataBinding.inflate(inflater, container, false)
        val root: View = binding.root

        val textViewTitle: TextView = binding.textViewTitle
        val textViewDescription: TextView = binding.textViewDescription
        dataViewModel.textViewTitle.observe(viewLifecycleOwner) {
            textViewTitle.text = it
        }
        dataViewModel.textViewDescription.observe(viewLifecycleOwner) {
            textViewDescription.text = it
        }
        return root
    }

}